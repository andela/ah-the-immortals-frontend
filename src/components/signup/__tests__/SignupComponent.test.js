import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SignupModal from '../SignupComponent';

describe('Tests signup component', () => {
  const props = {
    show: false,
    closeModal: jest.fn(),
    handleSubmit: jest.fn(),
    handleChange: jest.fn(),
    facebook: jest.fn(),
    google: jest.fn(),
    twitter: jest.fn(),
    handleSignInLink:jest.fn(),
    signupdata:{},
    errorShow:{}
  };

  it('Tests signup modal', () => {
    const wrapper = shallow(<SignupModal {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
