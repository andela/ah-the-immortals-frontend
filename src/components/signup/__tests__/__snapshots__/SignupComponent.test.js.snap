// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests signup component Tests signup modal 1`] = `
<Bootstrap(Modal)
  onHide={[MockFunction]}
  show={false}
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
  />
  <ModalBody>
    <div
      className="shadow-lg p-3 mb-4 bg-light rounded form-div"
    >
      <p>
        Sign Up
      </p>
      <Form
        errorShow={Object {}}
        handleChange={[MockFunction]}
        handleSubmit={[MockFunction]}
        signupdata={Object {}}
      />
      <h2>
        <span
          className="bg-light"
        >
          OR
        </span>
      </h2>
      <div
        className="round-icon"
      >
        <SocialComponent
          facebook={[MockFunction]}
          google={[MockFunction]}
          twitter={[MockFunction]}
        />
      </div>
    </div>
    <div
      className="link-text"
    >
      <p>
        Have an account?Â 
        <Link
          onClick={[MockFunction]}
          to="/login"
        >
          Sign In
        </Link>
      </p>
    </div>
  </ModalBody>
</Bootstrap(Modal)>
`;
